config/prod.secret.exs

```diff
@@ -1,29 +0,0 @@
-# In this file, we load production configuration and secrets
-# from environment variables. You can also hardcode secrets,
-# although such is generally not recommended and you have to
-# remember to add this file to your .gitignore.
-use Mix.Config
-
-secret_key_base =
-  System.get_env("SECRET_KEY_BASE") ||
-    raise """
-    environment variable SECRET_KEY_BASE is missing.
-    You can generate one by calling: mix phx.gen.secret
-    """
-
-config :document_viewer, DocumentViewerWeb.Endpoint,
-  http: [
-    port: String.to_integer(System.get_env("PORT") || "4000"),
```

Comment:
It looks like this is the issue: You need to include this line (or something similar like in `dev.exs`) to configure the port that the server is listening on.
Additionally, you need to add `server: true` to the endpoint configuration so that starting the application also starts the HTTP server.

Reply:
@paulswartz Thank you! My debugging efforts weren't reaching back far enough. There was an existing prod endpoint config, but I fixed it up and crucially it was missing the `server: true`.